### ISSUE TITLE
Error when using Vite (Vue 3)

### ISSUE THREAD

**Issue by lukef** [contentful/contentful.js#435]
## Expected Behavior

There is an error running applications using Vite (https://github.com/vitejs/vite). After installing the dependency using `yarn add contentful` and then running `yarn dev`, the server should start normally but fails with an error.

## Actual Behavior

Running `yarn dev` will cause the server to error with the following error:

```
[vite] Dep optimization failed with error:
'platform' is not exported by node-resolve:empty.js, imported by node_modules/contentful-sdk-core/dist/index.es-modules.js
Error: 'platform' is not exported by node-resolve:empty.js, imported by node_modules/contentful-sdk-core/dist/index.es-modules.js
    at error (<project dir>/node_modules/rollup/dist/shared/rollup.js:5231:30)
    at Module.error (<project dir>/node_modules/rollup/dist/shared/rollup.js:9787:16)
    at handleMissingExport (<project dir>/node_modules/rollup/dist/shared/rollup.js:9676:28)
    at Module.traceVariable (<project dir>/node_modules/rollup/dist/shared/rollup.js:10177:24)
    at ModuleScope.findVariable (<project dir>/node_modules/rollup/dist/shared/rollup.js:8708:39)
    at FunctionScope.findVariable (<project dir>/node_modules/rollup/dist/shared/rollup.js:2781:38)
    at ChildScope.findVariable (<project dir>/node_modules/rollup/dist/shared/rollup.js:2781:38)
    at Identifier$1.bind (<project dir>/node_modules/rollup/dist/shared/rollup.js:4137:40)
    at CallExpression$1.bind (<project dir>/node_modules/rollup/dist/shared/rollup.js:2867:23)
    at CallExpression$1.bind (<project dir>/node_modules/rollup/dist/shared/rollup.js:6605:15) {
  code: 'MISSING_EXPORT',
  url: 'https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module',
  id: '<project dir>/node_modules/contentful-sdk-core/dist/index.es-modules.js',
  pos: 122,
  loc: {
    file: '<project dir>/node_modules/contentful-sdk-core/dist/index.es-modules.js',
    line: 4,
    column: 9
  },
  frame: "2: import qs from 'qs';\n" +
    "3: import isPlainObject from 'lodash/isPlainObject';\n" +
    "4: import { platform, release } from 'os';\n" +
    '            ^\n' +
    '5: \n' +
    '6: function _defineProperty(obj, key, value) {',
   <watch section omitted>
```

## Possible Solution

The error directs you to the following page: https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module

## Steps to Reproduce

1. `yarn create vite-app someproject`
2. `cd someproject`
3. `yarn add contentful`
4. `yarn install`
5. `yarn dev`

## Context

Vite seems to be (?) the recommended CLI tool for Vue 3 development. We are having trouble proting our app to vue 3 due to this error.

## Environment

-   **Vite**: `v1.0.0-rc.4`
-   **Vue**: `v3.0.0`
-   **Language Version**: v14.5.0
-   **Package Manager Version**: yarn 1.22.4
-   **Operating System**: Linux 64 bit (Ubuntu 20.04)
-   **Package Version**: 7.14.8

**Comment by lukef** [comment]
It looks like this is reported here actually: https://github.com/contentful/contentful-sdk-core/issues/58

**Comment by princed** [comment]
A workaround for now: 

```ts
export default defineConfig({
  resolve: {
    alias: {
      // https://github.com/contentful/contentful-sdk-core/issues/58#issuecomment-675052926
      contentful: 'contentful/dist/contentful.browser',
    },
  },
  // ...
});

**Comment by marcolink** [comment]
@lukef the newly released version `8.1.8` should handle the browser case also for es modules. 
Please let me know if this doesn't fix your problem!

**Comment by princed** [comment]
@marcolink Unfortunately https://github.com/contentful/contentful.js/pull/625 doesn't fix the problem. 

So ES modules variant imports `contentful-sdk-core` directly, which is great. But the problem that `contentful-sdk-core` in its turn unconditionally imports Node.js builtins (`import { platform, release } from 'os'`) as mentioned in https://github.com/contentful/contentful-sdk-core/issues/58.

I can see two options here: 
1. Change `contentful-sdk-core` in way that it doesn't import `os` (at least unconditionally).
2. Create separately browser-friendly ES modules distribution, which less ideal because end users would still likely have to "target" manually similar to what I suggested above https://github.com/contentful/contentful.js/issues/435#issuecomment-786348416.


Do you think it's something you could do?

**Comment by marcolink** [comment]
@lukef @princed 
A new fix to the problem has been released! Please try version [8.5.1](https://github.com/contentful/contentful.js/releases/tag/v8.5.1). it's currently only on the `next` channel.

``` 
npm install contentful@next
```

**Comment by princed** [comment]
I've just updated to `9.0.0`, works like a charm. Thank you @marcolink!
